### PR TITLE
Solitaire: Correct scoring issues and invisible cards

### LINKED ISSUE #7991: Solitaire: Foundation stacks are not redrawn when <tab> moves multiple
See here:

image

A game started with 2 aces visible, I pressed <tab>, and both are moved to the foundation but one is invisible. Interacting with it makes it visible again.

### PR DESCRIPTION
**Pull request by AtkinsSJ**
We were not updating the score correctly in a couple of places:
- Undoing a FlipCard move
- Anything done by auto_move_eligible_cards_to_stacks()

FlipCard scoring is now handled by score_flip(), which imitates score_move().

Auto-moves using <tab> and those done with double-click now use the same code path to hopefully make them less error-prone.

This fixes #7991 too, where an auto-move that affected multiple cards would leave all the moved cards except the last one invisible.

### CODE REVIEW

File: Userland/Games/Solitaire/Game.cpp [hunk 1]
```diff
@@ -339,12 +339,13 @@ void Game::move_card(CardStack& from, CardStack& to)
     auto card = from.pop();
 
     card->set_moving(true);
-    m_focused_cards.clear();
     m_focused_cards.append(card);
```
Comment (trflynn89) [inline Userland/Games/Solitaire/Game.cpp:342]: I kinda think we can get rid of all three of these lines:
```c++
    card->set_moving(true);
    m_focused_cards.clear();
    m_focused_cards.append(card);
```

`move_card` is only called when moving a card to a foundation stack. Semantically, when a card is moved to the foundation this way, I would say it is not "focused" - I *think* focused cards are meant to be the card(s) actively being dragged, and its purpose is to make sure dragged cards are painted on top of everything else.

Removing all three of these lines seems to work fine on my end, could you try that as well?

(As an aside, I actually wouldn't be opposed to `move_cards` becoming a lambda in the new `attempt_to_move_card_to_foundations` to ensure its only used for moving cards to a foundation)

File: Userland/Games/Solitaire/Game.cpp [hunk 2]
```diff
@@ -152,6 +152,11 @@ void Game::score_move(CardStack& from, CardStack& to, bool inverse = false)
     }
 }
 
+void Game::score_flip(bool inverse = false)
```
[the remark this answers is not in the document]
Comment (AtkinsSJ) [reply Userland/Games/Solitaire/Game.cpp:156]: Oops! I don't know why I did that.

File: Userland/Games/Solitaire/Game.cpp (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Games/Solitaire/Game.cpp:342]: Thanks for the feedback! Sure, I'll give that a go. I wasn't clear about what focused meant - I might change its name if I can identify it for sure.

**Comment by AtkinsSJ** [comment]
@trflynn89 I did things a little differently than you suggested: https://github.com/SerenityOS/serenity/pull/7993/commits/3ec787347e34dd644f6a4cdd2fa3d88edb109300 - I'm interested in your thoughts there. Completely fine if you don't like it! Still trying to figure out the style here. Thank you for your help! :^)

**Review by trflynn89 — APPROVED** [review]

**Comment by trflynn89** [comment]
Cool, that last commit seems fine to move. The code to move the cards is now in a method named `attempt_to_move_card_to_foundations`, so it seems clear enough to me.

**Pull request merged**